### PR TITLE
Transfer filter tractogram

### PR DESCRIPTION
**Pull request by frheault**
Transfert of the scil_robust_tractogram_filtering_sequential.py script
- Support metadata
- Support TRK, TCK, VTK, etc. using StatefulTractogram
- Remove the notFiltered option (will use streamlines_math in the future)

### CODE REVIEW

File: scripts/scil_filter_tractogram.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+                   help='Slice number in Z, in voxel space')
+    p.add_argument('--filtering_list',
+                   help='Text file containing one rule per line\n'
+                   '(i.e. drawn_roi mask.nii.gz both_ends include)')
+    p.add_argument('--no_empty', action='store_true',
+                   help='Do not write file if there is no streamline')
+    p.add_argument('--output_not_filtered',
+                   help='Path to the remaining non filtered streamlines')
+    p.add_argument('-v', action='store_true', dest='verbose',
+                   help='Print the filtering information')
```
Comment (arnaudbore) [inline scripts/scil_filter_tractogram.py]: Add . at the end of each help sentence

File: scripts/scil_filter_tractogram.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                                description=DESCRIPTION)
+
+    p.add_argument('in_tractogram', metavar='IN_TRACTOGRAM',
+                   help='Path of the input tractogram file (.trk)')
+    p.add_argument('out_tractogram', metavar='OUT_TRACTOGRAM',
+                   help='Path of the output tractogram file (.trk)')
+
+    p.add_argument('--reference',
+                   help='Reference anatomy for Tck/Vtk file support')
+
```
Comment (arnaudbore) [inline scripts/scil_filter_tractogram.py]: Should we use the new add_reference from https://github.com/scilus/scilpy/pull/25

File: scripts/scil_filter_tractogram.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+            data_per_streamline[key] = [
+                sft.data_per_streamline[key][int(i)]
+                for i in indexes]
+
+        data_per_point = {}
+        for key in sft.data_per_point.keys():
+            data_per_point[key] = [
+                sft.data_per_point[key][int(i)]
+                for i in indexes]
+
```
Comment (arnaudbore) [inline scripts/scil_filter_tractogram.py]: Try to remove all the for loops

File: scripts/scil_filter_tractogram.py [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    - CRITERIA must be one of these values: ['include', 'exclude']
+
+    Multiple filtering tuples can be used and options mixed.
+    A logical AND is the only behavior available. All theses filtering
+    conditions will be sequentially applied.
+"""
+
+
+def _buildArgsParser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
```
Comment (arnaudbore) [inline scripts/scil_filter_tractogram.py]: should be _build_args_parser maybe we should agree on one name for all.
One name to rule them all !

File: scripts/scil_filter_tractogram.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+                    is_not)
+            elif geometry == 'Cuboid':
+                filtered_streamlines, indexes = filter_cuboid(
+                    sft,
+                    radius,
+                    center,
+                    filter_mode,
+                    is_not)
+
+        logging.debug('The filtering options %s resulted in '
```
Comment (GuillaumeTh) [inline scripts/scil_filter_tractogram.py]: use .format

File: scripts/scil_filter_tractogram.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+        data_per_streamline = sft.data_per_streamline[indexes]
+        data_per_point = sft.data_per_point[indexes]
+
+        sft = StatefulTractogram(filtered_streamlines, sft, Space.RASMM,
+                                 data_per_streamline=data_per_streamline,
+                                 data_per_point=data_per_point)
+
+    if not filtered_streamlines:
+        if args.no_empty:
+            logging.debug("The file %s won't be written (0 streamline)",
```
Comment (GuillaumeTh) [inline scripts/scil_filter_tractogram.py]: use .format

File: scripts/scil_filter_tractogram.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+                          args.out_tractogram)
+            return
+        else:
+            save_tractogram(sft, args.out_tractogram)
+            logging.debug("The file %s contains 0 streamline",
```
Comment (GuillaumeTh) [inline scripts/scil_filter_tractogram.py]: use .format

**Pull request merged**